drivers/sensor/mcp9600/mcp9600.h

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ * Copyright (c) 2020 SER Consulting LLC, Steven Riedl
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_MCP9600_MCP9600_H_
+#define ZEPHYR_DRIVERS_SENSOR_MCP9600_MCP9600_H_
+
+#include <errno.h>
+
+#include <zephyr/types.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <sys/util.h>
+#include <drivers/gpio.h>
+
+#define MCP9600_REG_TEMP_HOT		0x00
+
+#define MCP9600_REG_TEMP_DIFF		0x01
+#define MCP9600_REG_TEMP_COLD		0x02
+#define MCP9600_REG_RAW_ADC			0x03
+
+#define MCP9600_REG_STATUS			0x04
+
+#define MCP9600_REG_TC_CONFIG		0x05
+#define MCP9600_REG_DEV_CONFIG		0x06
+#define MCP9600_REG_A1_CONFIG		0x07
+
+#define MCP9600_REG_A2_CONFIG		0x08
+#define MCP9600_REG_A3_CONFIG		0x09
+#define MCP9600_REG_A4_CONFIG		0x0A
+#define MCP9600_A1_HYST				0x0B
+
+#define MCP9600_A2_HYST				0x0C
+#define MCP9600_A3_HYST				0x0D
+#define MCP9600_A4_HYST				0x0E
+#define MCP9600_A1_LIMIT			0x0F
+
+#define MCP9600_A2_LIMIT			0x10
+#define MCP9600_A3_LIMIT			0x11
+#define MCP9600_A4_LIMIT			0x12
+#define MCP9600_REG_ID_REVISION		0x13
+
+
+#define MCP9600_CFG_ALERT_ENA		BIT(0)
+#define MCP9600_CFG_ALERT_MODE_INT	BIT(1)
+#define MCP9600_CFG_ALERT_HI_LO		BIT(2)
+#define MCP9600_CFG_ALERT_RIS_FAL	BIT(3)
+#define MCP9600_CFG_ALERT_TH_TC		BIT(4)
+#define MCP9600_CFG_INT_CLEAR		BIT(7)
+
+struct mcp9600_data {
+	const struct device *i2c_master;
+
+	int16_t ttemp;
```

Comment:
This ordering wastes memory due to padding between the `int16_t` and `uint32_t` members.  Probably best to group them by whole and fraction.  Or simply store the value in an `int32_t` in some units that cover the sensor range and resolution, e.g. mCel, or what appears to be the native resolution: Cel/16.

Otherwise please document the relationship between the whole and fractional values with respect to sign and units.  I.e. is the true temperature `dtemp + dfrac / 1000000`?

Reply:
Actually am not using the other variables, so they were a waste anyway, removed